core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -939,7 +939,7 @@ private[spark] class TaskSetManager(
         && !isZombie) {
       for ((tid, info) <- taskInfos if info.executorId == execId) {
         val index = taskInfos(tid).index
-        if (successful(index) && !killedByOtherAttempt.contains(tid)) {
+        if (successful(index) && !info.running && !killedByOtherAttempt.contains(tid)) {
```

Comment:
let's add some code comments here to explain what's going on. e.g. we may have a running task whose partition has been marked as successful.

Reply:
ok, tks!